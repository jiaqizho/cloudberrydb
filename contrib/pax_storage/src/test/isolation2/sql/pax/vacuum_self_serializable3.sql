-- @Description Ensures that a vacuum with serializable works ok
-- 
DROP TABLE IF EXISTS pax_tbl;
DROP TABLE IF EXISTS pax_tbl2;
CREATE TABLE pax_tbl (a INT, b INT);
CREATE TABLE pax_tbl2 (a INT);
INSERT INTO pax_tbl SELECT i as a, i as b FROM generate_series(1, 100) AS i;

DELETE FROM pax_tbl WHERE a <= 30;
SET SESSION CHARACTERISTICS AS TRANSACTION ISOLATION LEVEL SERIALIZABLE;
SELECT COUNT(*) FROM pax_tbl;
SELECT ptblockname, pttupcount FROM get_pax_aux_table_all('pax_tbl');
VACUUM pax_tbl;
-- in case there's autovacuum worker running in the backend, the pax_tbl will not be dropped which has state = 2
SELECT ptblockname, pttupcount FROM get_pax_aux_table_all('pax_tbl') where pttupcount > 0;
SELECT COUNT(*) FROM pax_tbl;
