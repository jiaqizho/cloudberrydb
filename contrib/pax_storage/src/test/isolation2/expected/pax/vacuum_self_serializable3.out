-- @Description Ensures that a vacuum with serializable works ok
--
DROP TABLE IF EXISTS pax_tbl;
DROP
DROP TABLE IF EXISTS pax_tbl2;
DROP
CREATE TABLE pax_tbl (a INT, b INT);
CREATE
CREATE TABLE pax_tbl2 (a INT);
CREATE
INSERT INTO pax_tbl SELECT i as a, i as b FROM generate_series(1, 100) AS i;
INSERT 100

DELETE FROM pax_tbl WHERE a <= 30;
DELETE 30
SET SESSION CHARACTERISTICS AS TRANSACTION ISOLATION LEVEL SERIALIZABLE;
SET
SELECT COUNT(*) FROM pax_tbl;
 count 
-------
 70    
(1 row)
SELECT ptblockname, pttupcount FROM get_pax_aux_table_all('pax_tbl');
 ptblockname | pttupcount 
-------------+------------
 0           | 25         
 0           | 37         
 0           | 38         
(3 rows)
VACUUM pax_tbl;
VACUUM
-- in case there's autovacuum worker running in the backend, the pax_tbl will not be dropped which has state = 2
SELECT ptblockname, pttupcount FROM get_pax_aux_table_all('pax_tbl') where pttupcount > 0;
 ptblockname | pttupcount 
-------------+------------
 0           | 25         
 0           | 38         
 0           | 37         
(3 rows)
SELECT COUNT(*) FROM pax_tbl;
 count 
-------
 70    
(1 row)
